benchmarks/experiment_runner.py

```diff
@@ -242,41 +271,57 @@ def run_once_and_gather_metrics(self, benchmark_experiment, benchmark_model,
     if benchmark_experiment.xla:
       t_trace = 0
 
-    def loop(pytorch_profile=None):
+    def loop(pytorch_profile=None, iter_fn=None):
       nonlocal t_trace
+      total_timing = 0
       for i in range(self._args.iterations_per_run):
-
-        # Invoke iteration function and measure tracing time w/o waiting on the
-        # result.
-        if benchmark_experiment.xla:
-          t_trace_start = time.perf_counter()
-        output = benchmark_model.model_iter_fn(
-            inputs_list[i], collect_full_output=self._args.collect_full_output)
-        if benchmark_experiment.xla:
-          t_trace += time.perf_counter() - t_trace_start
+        output, timing, trace = iter_fn(benchmark_experiment, benchmark_model,
+                                        inputs_list[i])
+        if trace is not None:
+          t_trace += trace
+        if timing is not None:
+          total_timing += timing
 
         # Mark step.
         self._mark_step(benchmark_experiment)
         if pytorch_profile is not None:
           pytorch_profile.step()
 
       self._synchronize(benchmark_experiment)
-      return output
+      return output, total_timing
 
     # Execute all iterations (with) profiling.
     enable_pytorch_profiling = self._args.dump_pytorch_profiles or \
         self._args.profile_cuda_cpu or \
         self._args.profile_cuda_cpu_individual_ops
     if enable_pytorch_profiling:
+      if self._args.pure_wall_time:
+        logger.warning(
+            'Run with pure wall time, but also with profiling flags enabled. Falling back to a default wall time.'
+        )
       with profile(activities=[ProfilerActivity.CUDA, ProfilerActivity.CPU
                               ]) as pytorch_profile:
-        output = loop(pytorch_profile)
+        output, _ = loop(pytorch_profile, iter_fn=self._default_iter_fn)
     else:
-      output = loop()
+      if self._args.pure_wall_time:
+        output, pure_wall_timing = loop(iter_fn=self._pure_wall_time_iter_fn)
```

Comment:
bench.py is intended for benchmarking on CUDA, please prohibit the TPU xla routing from entering the `_pure_wall_time_iter_fn`.